api/src/controllers/export-data.js

```diff
@@ -101,17 +88,25 @@ module.exports = {
       })
       .then((userCtx) => {
         if (!auth.hasAllPermissions(userCtx, 'can_export_all')) {
-          return auth.check(req, getExportPermission(req.params.type));
+          return auth.check(req, service.permission(type));
         }
       })
       .then(() => {
-        writeExportHeaders(res, req.params.type, formats.csv);
+        const format = service.format(type);
+        writeExportHeaders(res, type, format);
+
+        const writeAsStream = format === 'csv';
+        if (!writeAsStream) {
+          return service.exportObject(type, filters, options)
+            .then(obj => res.json(obj));
```

Comment:
If we want to support human readable, should we also have this "humanly" format the JSON? (multiple lines, tabbed out etc).

Reply:
Since our final design ended up prioritizing a "medic app" inside DHIS2, I think a lot of the human readability requirements can be handled there instead of through this feature. I implemented it since it is provided for other `export` endpoints, and because it is quite helpful for testing larger datasets (for example, once we start building the 75+ indicators for the MoH 515). The indenting is easy enough to get through an IDE or online tool.